daemon/buildinfo.c

```diff
@@ -200,8 +200,67 @@
 
 #define FEAT_YES_NO(x) ((x) ? "YES" : "NO")
 
+// coverity[ +tainted_string_sanitize_content : arg-0 ]
+static inline void coverity_remove_taint(char *s)
+{
+    (void)s;
+}
+
+void read_install_type(void) {
+    char *filepath;
+    filepath = mallocz(sizeof(char) * (strlen(netdata_configured_user_config_dir) + strlen(".install-type") + 2));
+    sprintf(filepath, "%s/%s", netdata_configured_user_config_dir, ".install-type");
+    FILE *fp = fopen(filepath, "r");
+
+    if(fp) {
+        char *line = mallocz(sizeof(char) * (200 + 1));
+        while (fgets(line, 200, fp) != NULL) {
+            char *value = line;
+
+            while (*value && *value != '=') value++;
+
+            if (*value == '=') {
+                *value = '\0';
+                value++;
```

Comment:
what if the next byte after the `=` is `\0` e.g. file ends after `=` without newline like with `echo -n INSTALL_TYPE=`

Reply:
It should, but I had not thought of that.

Note that the same case is also incorrectly handled when we’re parsing the output of the `system-info.sh` script (among other issues there, I will probably be doing a PR to improve things there later).